SofaKernel/framework/sofa/core/objectmodel/BaseData.h

```diff
@@ -78,13 +78,13 @@ class SOFA_CORE_API BaseData : public DDGNode
     public:
         BaseInitData() : data(nullptr), helpMsg(""), dataFlags(FLAG_DEFAULT), owner(nullptr), name(""), ownerClass(""), group(""), widget("") {}
         BaseData* data;
-        const char* helpMsg;
+        std::string helpMsg;
         DataFlags dataFlags;
         Base* owner;
-        const char* name;
-        const char* ownerClass;
-        const char* group;
-        const char* widget;
+        std::string name;
+        std::string ownerClass;
+        std::string group;
+        std::string widget;
```

Comment:
are all those info of the Data only used for gui?

Reply:
> Error from CI :
> 
> ```
> Binding_Data.cpp:866:35: error: cannot convert ‘const string {aka const std::__cxx11::basic_string<char>}’ to ‘const char*’ in initialization
>      const char *h = data->getHelp() ;
> ```
> 
> Could you also give explanation for this change @marques-bruno ?

It's a breaking change and I don't compile with the same settings as the CI. I'll fix those issues as they show up.
Regarding the "why":

I had some segmentation faults with strings that were badly set in BaseData::group and since char* is so 80's I thought it's the occasion to replace them with std::strings, like a normal c++ developer does :)
It solved my problem, hence the PR

Reply:
> are all those info of the Data only used for gui?

name, help and group are intensively used in SofaPython and SofaPython3. OwnerClass could also be. widget is probably exclusively used by the GUI

Reply:
This change will also fix #1092 :+1:

Reply:
:warning: Also, as stated in #1092, don't forget to revert #1090 (temporary fix).

Reply:
> warning Also, as stated in #1092, don't forget to revert #1090 (temporary fix).

The temp fix in #1090 is overridden by the PR already, so nothing to revert :+1: